### DISCUSSION TITLE
How to apply an initial transformation

### DISCUSSION THREAD

**Discussion by bercowsky** [SuperElastix/elastix#1141] [Q&A]
I am trying to register 2 images using rigid registration, but I want to apply an initial transformation before hand. I saw there is the `-t0` option in the command line, but is there a way to inform elastix about this transformation in C++?

**Comment by N-Dekker** [comment]
Thanks for your question, @bercowsky Do you use the `ElastixRegistrationMethod`? If so, you may call its `SetInitialTransform` member function, and specify the initial transform as argument:

https://github.com/SuperElastix/elastix/blob/fa93025ddf78c61148882882eb606ab93eb67eaf/Core/Main/itkElastixRegistrationMethod.h#L218-L220

**Comment by bercowsky** [comment]
Thanks for your reply! I am trying to register the images using `ELASTIX::RegisterImages` as shown in the code below, but I have not found an easy way to apply the initial transformation from my code.

```cpp
elastix->RegisterImages(
        static_cast<typename itk::DataObject::Pointer>(_FixITKImage),
        static_cast<typename itk::DataObject::Pointer>(_MovingITKImage),
        parameters, // Parameter map read in previous code
        _OutputDir, // Directory where output transform params are written, if
                    // enabled
        elastixLogs, // Enable/disable writing of elastix.log
        masking
            ? static_cast<typename itk::DataObject::Pointer>(_FixITKMaskImage)
            : nullptr, // Provide fixed image mask (optional, 0 = no mask)
        nullptr // Provide moving image mask (optional, 0 = no mask)
    );
```
